gateway/src/apicast/proxy.lua

```diff
@@ -231,11 +241,11 @@ function _M:call(service)
 
   return function()
     -- call access phase
-    return self:access(service)
+    return self:access(service, context)
   end
 end
 
-function _M:access(service)
+function _M:access(service, context)
```

Comment:
@mikz  I decided to pass the context instead of passing just the usage to be increased because in the future there might be other policies passing other information in the context. This way we won't need to change the signature of the access() method again.

Reply:
I'm not sure. https://github.com/3scale/apicast/pull/556/files#diff-31313c92616b54028dac1ba183e6f79aR287 was all the info it needed. It is quite a lot of params, but not that bad imo.

Reply:
I'm OK with adding 2 or 3 extract params. I'm more concerned about changing the signature of `access()` in the future and breaking backwards compatibility.

Reply:
I agree. Maybe we can build on https://github.com/3scale/apicast/pull/556/ and try to figure out out there and then just use it from here?

Reply:
Sounds good :+1: